File: src/Java/CreationalPatterns/AbstractFactory/EraStyles/src/ConcreteFactories/HouseFactory.java
package ConcreteFactories;

import Abstraction.EraObject;
import Abstraction.EraObjectStylesFactory;
import Products.MedievalHouse;
import Products.RenaissanceHouse;
import Products.VictorianHouse;

public class HouseFactory extends EraObjectStylesFactory {
    @Override
    public EraObject createMedievalStyleObject() {
        return new MedievalHouse();
    }

    @Override
    public EraObject createRenaissanceStyleObject() {
        return new RenaissanceHouse();
    }

    @Override
    public EraObject createVictorianEraStyleObject() {
        return new VictorianHouse();
    }
}

File: src/C#/CreationalPatterns/AbstractFactory/EraStyles/ConcreteFactories/HouseFactory.cs
using EraStyles.Abstraction;
using EraStyles.Products;
using System;
using System.Collections.Generic;
using System.Text;

namespace EraStyles.ConcreteFactories
{
    public class HouseFactory : EraObjectStylesFactory
    {
        public override EraObject CreateMedievalStyleObject()
        {
            return new MedievalHouse();
        }

        public override EraObject CreateRenaissanceStyleObject()
        {
            return new RenaissanceHouse();
        }

        public override EraObject CreateVictorianEraStyleObject()
        {
            return new VictorianHouse();
        }
    }
}
